### ISSUE TITLE
hifiasm+ULont assembly results

### ISSUE THREAD

**Issue by jlipovac** [chhylp123/hifiasm#357]
Hi!

First, thanks for the great tool! :)
I am assembling two different genomes - the human genome and the durian (fruit) genome. 

For human genome I have around 48x pacbio hifi data and around 23x ULont data. I also have around 23x paternal and 28x maternal pacbio hifi data. Firstly, I ran hifiasm + trio binning. I mapped the obtained assembly for haplotype 1 to the chm13 reference and with paftools asmstat option I got NG50 59.6Mbp and NGA50 50.8Mbp. After that I ran hifiasm + trio binning + ULont and for haplotype 1 got NG50 53.4Mbp and NGA50 48.3Mbp. The same thing (decrease in NG50 and NGA50) happened with haplotype 2. What do you think, what would be the explanation for the decrease in NGA50 and NG50 measures after adding ULont reads?

In case of durian I have 60x pacbio hifi, 30x ULont and 60x Omni-C data. Firstly I ran hifiasm + hic, did the same thing, mapped haplotype 1 to the reference and with paftools asmstat got NG50 20.GMbp. After repeated assembly, but this time with ULont reads, I got an identical NG50. Same thing for haplotype2. I don't know what could be the cause of that and I'm interested in your thoughts.

Thanks a lot in advance!

Josipa

**Comment by chhylp123** [comment]
What is the N50 of the ONT-integrated assembly?  Is it increased or not? For the NGA50, it highly depends on the alignment. The ONT-integration assembly can assemble repeat-rich regions, but these regions are not easily aligned.

**Comment by jlipovac** [comment]
N50 of the human genome ONT-integrated assembly is decreased, and of the durian ONT-integrated assembly remained the same.

**Comment by chhylp123** [comment]
Sounds like impossible? Are you using public data? Could you please let me know the link?

**Comment by jlipovac** [comment]
Sorry, the data is not public, but I can share durian data with you. Do you have any suggestions on how best to share them with you?

**Comment by chhylp123** [comment]
Thanks a lot. Do you have ONT/Onedrive?

**Comment by likui345** [comment]
Hi!
After adding ont data, I found many lines containing ‘scaf’ in the gfa files. The number of lines seems irregular. Do these lines affect the assembly accuracy? Can these lines be avoided? Here are some examples：
A       ptg000004l      9020521 -       scaf    0       356     id:i:2186467    HG:A:a
A       ptg000006l      7056282 +       scaf    0       356     id:i:2186474    HG:A:a

Thanks a lot ！

**Comment by chhylp123** [comment]
Which version are you using? The latest version shouldn't have any or very few `scaf`.

**Comment by likui345** [comment]
> Which version are you using? The latest version shouldn't have any or only has very few `scaf`.
Yes, you are right. For the version 0.17.4-r455, there are many ‘scaf’ in the gfa file. However, for the version 0.17.7-r461, there are only two ‘scaf’.

**Comment by likui345** [comment]
> Which version are you using? The latest version shouldn't have any or only has very few `scaf`.

Yes, you are right. For the version 0.17.4-r455, there are many "scaf" in the gfa file. However, for the version 0.17.7-r461, there are only two "scaf".

**Comment by chhylp123** [comment]
Please always try the latest version: https://github.com/chhylp123/hifiasm/releases/tag/0.18.5. The latest version is much better than before. It would be better to rerun the latest version from the scratch.

**Comment by yangting662** [comment]
Has the issue been resolved? I'm having a similar problem. @chhylp123 @likui345 @jlipovac 

I am assembling an allohexaploid plant genome for a T2T assembly using HiFi (~47x), Hi-C (~100x), and UL ONT (~58x) data.

With HiFi + Hi-C, I obtained an assembly of about 2.27 Gb with N50 around 64 Mb. However, after adding UL ONT reads with -ul, the N50 decreased substantially. Using raw UL ONT, the N50 dropped to around 8.5 Mb. After correcting ONT reads with NECAT and using the corrected reads as -ul input, the N50 was around 16 Mb, still much lower than the HiFi + Hi-C assembly.

## ONT data statistics

The raw/pass UL ONT data used in this project are as follows:

| Flag | TotalBase | TotalReads | MaxLen | AvgLen | N50 | L50 | N90 | L90 | meanQ |
|---|---:|---:|---:|---:|---:|---:|---:|---:|---:|
| pass | 133,084,947,288 | 3,267,704 | 779,087 | 40,727.36 | 100,000 | 421,504 | 21,412 | 1,448,483 | 17.55 |

## Assembly results

The assembly results from different strategies are shown below:

| Assembly Strategy | num_seqs | sum_len | N50 |
|---|---:|---:|---:|
| HiFi + Hi-C, run 1 | 1,675 | 2,280,903,496 | 64,120,605 |
| HiFi + Hi-C, run 2 | 1,706 | 2,262,900,936 | 64,131,239 |
| HiFi + Hi-C + raw UL ONT | 1,541 | 2,045,662,242 | 8,517,863 |
| HiFi + Hi-C + NECAT-corrected UL ONT | 3,034 | 2,339,029,476 | 16,179,424 |


My commands were (hifiasm 0.25.0-r726) :
hifiasm -o genome.asm -t 32 --n-hap 6 --h1 hic_1.fq.gz --h2 hic_2.fq.gz hifi_reads.fastq  # HiFi + Hi-C

hifiasm -o genome.asm -t 32 --n-hap 6  --h1 hic_1.fq.gz --h2 hic_2.fq.gz  -ul ont.pass.fastq.gz hifi_reads.fastq  # HiFi + Hi-C + raw UL ONT

hifiasm -o genome.asm -t 32 --n-hap 6 --h1 hic_1.fq.gz --h2 hic_2.fq.gz -ul cns_final.fasta hifi_reads.fastq  # HiFi + Hi-C + NECAT-corrected UL ONT

I would like to ask:

Why might adding UL ONT reads decrease N50 so strongly in this allohexaploid genome? Could homoeologous regions or repeats cause conflicting ONT long-range signals?
Should ONT reads be preprocessed, length-filtered, or corrected before being used with -ul?
Is raw ONT FASTQ preferable to NECAT-corrected ONT FASTA for -ul?
Are there recommended parameters for combining HiFi + Hi-C + UL ONT in a polyploid genome, especially with --n-hap 6?
Could the decrease in N50 indicate that ONT long-range signals are conflicting with the HiFi/Hi-C graph because of highly similar homoeologous regions?
In this situation, would you recommend using ONT only for local gap closing or validation instead of including it globally in hifiasm?

Thank you very much for your help.

yangting
